integration-tests-kapt/src/jmh/java/me/tatarka/inject/benchmark/ScopedAccessBenchmark.kt

```diff
@@ -0,0 +1,49 @@
+package me.tatarka.inject.benchmark
+
+import me.tatarka.inject.internal.lazyGet
+import org.openjdk.jmh.annotations.Benchmark
+import org.openjdk.jmh.annotations.Scope
+import java.util.concurrent.ConcurrentHashMap
+
+// @Inject @MyScope
+class Foo
+
+// @MyScope @Component
+abstract class ScopedAccessComponent {
+    abstract val foo: Foo
+}
+
+
+class InjectLazyScopedAccessComponent : ScopedAccessComponent() {
+    private val _foo by lazy {
+        Foo()
+    }
+
+    override val foo: Foo
+        get() = _foo
+}
+
+class InjectDynamicScopedAccessComponent : ScopedAccessComponent() {
+    private val _scope = ConcurrentHashMap<String, Any?>()
+
+    override val foo: Foo = _scope.lazyGet("me.tatarka.inject.benchmark.Foo") { Foo() }
+}
+
+open class ScopedAccessBenchmark {
```

Comment:
Comparing performance of field vs map impls. Seem to be the same order of magnitude.